api/api.go

```diff
@@ -417,6 +421,62 @@ type CloudBackupSchedEnumerateResponse struct {
 	Schedules map[string]CloudBackupScheduleInfo
 }
 
+type Operation string
+
+const (
+	MigrateCluster     = Operation("MigrateCluster")
+	MigrateVolume      = Operation("MigrateVolume")
+	MigrateVolumeGroup = Operation("MigrateVolumeGroup")
+)
+
+type MigrateStatus string
+
+const (
+	MigrateStatusQueued       = MigrateStatus("Queued")
+	MigrateStatusInitializing = MigrateStatus("Initializing")
+	MigrateStatusInProgress   = MigrateStatus("InProgress")
+	MigrateStatusFailed       = MigrateStatus("Failed")
+	MigrateStatusComplete     = MigrateStatus("Complete")
+)
+
+type MigrateStage string
+
+const (
+	MigrateStageBackup       = MigrateStage("Backup")
+	MigrateStageRestore      = MigrateStage("Restore")
+	MigrateStageVolumeUpdate = MigrateStage("VolumeUpdate")
+	MigrateStageDone         = MigrateStage("Done")
+)
+
+type CloudMigrateInfo struct {
+	ClusterId       string
+	LocalVolumeId   string
+	LocalVolumeName string
+	RemoteVolumeId  string
+	CloudSnapId     string
+	CurrentStage    MigrateStage
+	Status          MigrateStatus
+	LastUpdate      time.Time
+	LastSuccess     time.Time
+}
+
+type CloudMigrateStartRequest struct {
+	Operation Operation
+	ClusterId string
+	// Depending on Operation type this can be VolumeID or VolumeGroupID
+	TargetId string
+}
+
```

Comment:
might be a good idea to add migrateStartResponse structure that returns a unique ID? this unique ID can then be used to get status and also for cancelling the request?  With volume group migration( using volume labels), both status and cancel requests becomes easier with unique id.

Reply:
The way it is right now, you would need to give it the same targetID for starting and canceling the operation. Migrating the group would take in labels, and you would pass in labels to both. We can update these to return task IDs once we have corresponding task IDs for cloudsnaps.